ADR/0036-trusted-artifacts.md

```diff
@@ -0,0 +1,47 @@
+# 36. Trusted Artifacts
+
+Date: 2024-06-11
+
+## Status
+
+Accepted
+
+## Context
+
+One of the properties of Konflux is that users should be allowed to include their own Tekton Tasks
+in a build Pipeline, e.g. to execute unit tests, without invalidating the build process. This is
+distinct from other build systems where a rigid process prevents users from applying such
+customizations. To support this, Konflux build Pipelines use Trusted Artifacts to securely share
+data between Tasks. Enterprise Contract is then responsible for verifying that Trusted Artifacts
+were properly used in the parts of the build Pipeline that affect the build outcome, e.g.
+`git-clone` and `buildah` Tasks.
+
+Trusted Artifacts is inspired by the upcoming work being done by the Tekton Community,
```

Comment:
The upstream work will require changes to the Tekton resources IIUC. I don't think there's a way to avoid a migration. But hopefully, this is isolated to Task definitions and we can switch over without requiring a migration from the user side.